src/pages/user/UserInstitution.tsx

```diff
@@ -72,89 +72,121 @@ const UserInstitution: FC = () => {
-                    setValueFunction={inputValue => {
-                      setFieldValue(FormikInstitutionUnitFieldNames.NAME, inputValue.name);
-                      setFieldValue(FormikInstitutionUnitFieldNames.ID, inputValue.id);
-                      setFieldValue(name, inputValue ?? emptyRecursiveUnit);
-                    }}
-                    placeholder={t('organization.search_for_institution')}
-                  />
-                  {values.unit && (
-                    <>
-                      <InstitutionSelector counter={0} unit={value} />
-                      <StyledButton
-                        onClick={() => handleSubmit()}
-                        variant="contained"
-                        type="submit"
-                        color="primary"
-                        disabled={!values.unit}
-                        data-testid="institution-add-button">
-                        {t('common:add')}
-                      </StyledButton>
-                      <StyledButton
-                        onClick={() => {
-                          toggleOpen();
-                          resetForm({});
-                        }}
-                        variant="contained"
-                        color="secondary">
-                        {t('common:cancel')}
-                      </StyledButton>
-                    </>
-                  )}
-                </StyledInstitutionSearchContainer>
-              )
-            }
+            {({ field: { name, value } }: FieldProps) => (
+              <>
+                {!editMode && units.length > 0 ? (
+                  units.map((unit: FormikInstitutionUnit, index: number) => (
+                    <InstitutionCard key={index} unit={unit} onEdit={handleEdit} />
+                  ))
+                ) : (
+                  <>{!open && <NormalText>{t('organization.no_institutions_found')}</NormalText>}</>
+                )}
+                {open && (
+                  <StyledInstitutionSearchContainer>
+                    <InstitutionSearch
+                      dataTestId="autosearch-institution"
+                      disabled={editMode}
+                      label={t('organization.institution')}
+                      clearSearchField={values.name === ''}
+                      setValueFunction={inputValue => {
+                        setFieldValue(FormikInstitutionUnitFieldNames.NAME, inputValue.name);
+                        setFieldValue(FormikInstitutionUnitFieldNames.ID, inputValue.id);
+                        setFieldValue(name, inputValue ?? emptyRecursiveUnit);
+                      }}
+                      placeholder={editMode ? values.name : t('organization.search_for_institution')}
+                    />
+                    {values.unit && (
+                      <>
+                        <InstitutionSelector counter={0} unit={value} />
+                        <StyledButton
+                          onClick={() => handleSubmit()}
```

Comment:
trenger ikke denne når du allerede har `onSubmit` på Formik og `type="submit"` her

Reply:
Jo, jeg må ha med det for å kunne overskrive Formik sin onSubmit. Jeg vil bruke Formik sin funksjon for resetForm, men vil også gjøre litt mer greier. Har ikke funnet en annen måte å gjøre det på. Vet du om en annen måte å gjøre det på?

Reply:
jeg mener at 
```
<Formik onSubmit={myOnSubmit}>
   <Button type="submit" />
</Formik>
```
er det samme som
```
<Formik>
   <Button onClick={myOnSubmit} />
</Formik>
```
hvor den første versjonen også har div Formik props som resetForm osv. se `AdminCustomerInstitutionPage.tsx` feks

Reply:
ah, ok 👍  skal teste

Reply:
hvis jeg fjerner onClicken, så blir ikke onSubmit trigget. kan sjekke ut litt hvorfor det skjer

Reply:
ahhh - må tydeligvis ha `<Form>` tagen for at man kan bruke det sånn som du sier 💡

Reply:
åja, det var nytt for meg også